### ISSUE TITLE
Nan gradients with jaxoplanet.experimental.starry

### ISSUE THREAD

**Issue by shashankdholakia** [exoplanet-dev/jaxoplanet#235]
Testing `jaxoplanet.experimental.starry` with the assert statement in #234 commented out, I'm getting `nan` gradients for all values except ones associated with maps of either body. Here is the same MWE, but with the gradients:

```python
from jaxoplanet.experimental.starry.surface import Surface
from jaxoplanet.orbits.keplerian import Central
from jaxoplanet.experimental.starry.orbit import SurfaceSystem
from jaxoplanet.experimental.starry.ylm import Ylm
from jaxoplanet.experimental.starry.light_curves import light_curve

from jax import config
config.update("jax_enable_x64", True)

import numpy as np
import jax
from jax import vmap, jit
import jax.numpy as jnp
import matplotlib.pyplot as plt

def compute_light_curve(params, times):
    period = params['period']
    central_surface = Surface(inc=jnp.pi/2, obl=0.0, period=np.inf, u=params['u'], amplitude=1.0)
    central = Central(radius=params['r_star'], mass=params['m_star'])

    inc = jnp.radians(params['inc'])
    y = Ylm.from_dense(jnp.array([1.0, 0, params['y10'], 0.0]))
    body_surface = Surface(y=y, inc=inc, obl=0.01, period=period, amplitude=params['amplitude'], normalize=False)
    body = {
        "radius": params['r_body'],
        "mass": params['m_body'],
        "period": period,
        "inclination": inc,
        "surface": body_surface,
        "time_transit": params['time_transit'],
    }

    spheres = SurfaceSystem(central, central_surface).add_body(**body)
    return light_curve(spheres)(times).sum(axis=1)
  
params_dict = {
    'period': 1.0,
    'r_star': 1.0,
    'm_star': 1.0,
    'r_body': 0.1,
    'm_body': 0.01,
    'inc': 90.0,
    'u': jnp.array([0.2, 0.2]),
    'time_transit': 0.0,
    'y10': 0.5,
    'amplitude': 0.001,
}

times = jnp.linspace(-1, 1, 1000)

grad_compute_lc = jax.jacrev(compute_light_curve)
grads = grad_compute_lc(params_dict, times)
_, axes = plt.subplots(len(grads.keys()), 1, figsize=(6, 10), sharex=True)
for n, key in enumerate(grads.keys()):
    print(key)
    axes[n].plot(times, grads[key])
    axes[n].set_ylabel(key)
```
![jaxoplanet_grads](https://github.com/user-attachments/assets/f274b0b9-6817-4437-a20f-e91dd2374502)

**Comment by lgrcia** [comment]
Thanks so much for catching that! For reference this is due to a zero value in a square root of the starry `solution_vector` function. Using `utils.zero_safe_sqrt` fixed it! (merged!)
